### ISSUE TITLE
Rule proposal: method-is-parameter-maybe

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#5124]
Every function that has a name `/^is[A-Z]/` must have first parameter name `/^maybe[A-Z]`, e.g.

```js
let isReactComponent;

/**
 * Determines if the given object has the signature of a class that inherits React.Component.
 *
 * @param {*} Component
 * @returns {boolean}
 */
isReactComponent = (maybeReactComponent) => {
    return 'prototype' in maybeReactComponent && _.isFunction(maybeReactComponent.prototype.render);
};
```

**Comment by eslintbot** [comment]
@gajus Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
That seems awfully specific and (to my eyes) seems better as a custom rule or plugin rule.

**Comment by nzakas** [comment]
Agree with @platinumazure, this is best as a custom rule.
